core/src/sql/value/value.rs

```diff
@@ -2671,6 +2671,46 @@ impl Value {
 		}
 	}
 
+	/// Partially process this type (just the params) returning an evaluated simple Value that still needs computing
+	///
+	/// Is used recursively.
+	pub(crate) async fn partially_compute_unbordered(
+		&self,
+		stk: &mut Stk,
+		ctx: &Context<'_>,
+		opt: &Options,
+		doc: Option<&CursorDoc<'_>>,
+	) -> Result<Value, Error> {
+		// Prevent infinite recursion due to casting, expressions, etc.
+		let opt = &opt.dive(1)?;
+
+		match self {
+			// We only want to process params
+			Value::Param(v) => stk.run(|stk| v.compute(stk, ctx, opt, doc)).await,
+			// Anything that contains params gets partially computed
+			Value::Array(v) => stk.run(|stk| v.partially_compute(stk, ctx, opt, doc)).await,
```

Comment:
Perhaps objects too?

Reply:
Idiom paths can also contain variables

Reply:
`Value::Block` does not fall under `Value::Expression`, do we want to evaluate those too?

Reply:
I can actually keep commented but nearly all of these entries can contain variables 😅 
<img width="267" alt="image" src="https://github.com/surrealdb/surrealdb/assets/17960250/7e0b6245-23cf-45ef-bd5a-90b91acde79c">

Reply:
Yeah we can expand as needed

Reply:
I have added idiom, but left the rest intact